### PR TITLE
doc(k8s-mesh) updated documentation about k8s sidecar mesh

### PR DESCRIPTION
**Pull request by hutchic**
merge after https://github.com/Kong/docs.konghq.com/pull/1351

### CODE REVIEW

**Comment by levfishbluefish** [comment]
We will need to add a navigation item for this page -- suggesting we put it under "Guides and References" for now

**Comment by levfishbluefish** [comment]
> Add to navigation before merging

Ok apparently change this to "Add to hub instead"

File: app/1.2.x/kubernetes-and-service-mesh.md [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+## Introduction
+
+Kong `0.15.0` / `1.0.0` added the capability to proxy and route raw `tcp` and `tls`
+streams and deploy Kong using a service-mesh sidecar pattern with mutual
+`tls` between Kong nodes. This tutorial walks you through setting up a Kong service
+mesh on Kubernetes with our [Kubernetes sidecar injector plugin](https://github.com/Kong/kubernetes-sidecar-injector)
+
+## Prerequisites
+
+You need Kong 110.0 or later running on Kubernetes including and a ssl certificate
```
Comment (h3xar0n) [inline app/1.2.x/kubernetes-and-service-mesh.md]:
```suggestion
You need Kong 1.0.0 or later running on Kubernetes including and an SSL certificate
```

File: app/1.2.x/kubernetes-and-service-mesh.md [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+stored as a secret that's available to the Kong control plane. The Make tasks `run_cassandra`
+and `run_postgres` from the [Kong Kubernetes Repository](https://github.com/Kong/kong-dist-kubernetes)
+will fully provision the prerequisite data store, Kong control plane, Kong data plane
+and the ssl secret.
```
Comment (h3xar0n) [inline app/1.2.x/kubernetes-and-service-mesh.md]:
```suggestion
and the SSL secret.
```

File: app/1.2.x/kubernetes-and-service-mesh.md [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+
+Alternatively follow the any of the setup instructions from 
+[Kong Kubernetes Install Instructions](/install/kubernetes/) page then setup the 
+ssl certificate / secret
```
Comment (h3xar0n) [inline app/1.2.x/kubernetes-and-service-mesh.md]:
```suggestion
ssl certificate / secret
```
```suggestion
SSL certificate/secret:
```

File: app/1.2.x/kubernetes-and-service-mesh.md [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+  - key encipherment
+  - server auth
+EOF
+kubectl certificate approve kong-control-plane.kong.svc
+kubectl -n kong create secret tls kong-control-plane.kong.svc --key=privkey.pem --cert=<(kubectl get csr kong-control-plane.kong.svc -o jsonpath='{.status.certificate}' | base64 --decode)
+kubectl delete csr kong-control-plane.kong.svc
+rm privkey.pem
+```
+
+or use the following convenience script
```
Comment (h3xar0n) [inline app/1.2.x/kubernetes-and-service-mesh.md]:
```suggestion
or use the following convenience script:
```

File: app/1.2.x/kubernetes-and-service-mesh.md [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+
+```
+curl -fsSL https://raw.githubusercontent.com/Kong/kong-dist-kubernetes/master/setup_certificate.sh | bash
+```
+
+## Installation Steps
+
+Export some variables to access the Kong admin api and Proxy
```
Comment (h3xar0n) [inline app/1.2.x/kubernetes-and-service-mesh.md]:
```suggestion
Export some variables to access the Kong Admin API and Proxy:
```

File: app/1.2.x/kubernetes-and-service-mesh.md [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+
+```
+$ export HOST=$(kubectl get nodes --namespace default -o jsonpath='{.items[0].status.addresses[0].address}')
+$ export ADMIN_PORT=$(kubectl get svc --namespace kong kong-control-plane  -o jsonpath='{.spec.ports[0].nodePort}')
+```
+
+Enable the sidecar injector plugin via the Kong admin api
```
Comment (h3xar0n) [inline app/1.2.x/kubernetes-and-service-mesh.md]:
```suggestion
Enable the Sidecar Injector Plugin via the Kong Admin API:
```

File: app/1.2.x/kubernetes-and-service-mesh.md [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+
+```
+curl $HOST:$ADMIN_PORT/plugins -d name=kubernetes-sidecar-injector -d config.image=kong
+```
+
+Turn on Kubernetes sidecar injection
```
Comment (h3xar0n) [inline app/1.2.x/kubernetes-and-service-mesh.md]:
```suggestion
Turn on Kubernetes Sidecar Injection:
```

File: app/1.2.x/kubernetes-and-service-mesh.md [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+  clientConfig:
+    service:
+      namespace: kong
+      name: kong-control-plane
+      path: /kubernetes-sidecar-injector
+    caBundle: $(kubectl config view --raw --minify --flatten -o jsonpath='{.clusters[].cluster.certificate-authority-data}')
+EOF
+```
+
+or use the following convenience script
```
Comment (h3xar0n) [inline app/1.2.x/kubernetes-and-service-mesh.md]:
```suggestion
or use the following convenience script:
```

File: app/1.2.x/kubernetes-and-service-mesh.md [hunk 9]
```diff
@@ -0,0 +1,129 @@
...
+
+```
+curl -fsSL https://raw.githubusercontent.com/Kong/kong-dist-kubernetes/master/setup_sidecar_injector.sh | bash
+```
+
+## Usage
+
+Going forward any pod's that start get a Kong sidecar automatically injected and all
```
Comment (h3xar0n) [inline app/1.2.x/kubernetes-and-service-mesh.md]:
```suggestion
Going forward, any pods that start get a Kong Sidecar automatically injected, and all
```

File: app/1.2.x/kubernetes-and-service-mesh.md [hunk 10]
```diff
@@ -0,0 +1,129 @@
...
+data from the containers of that pod will flow through a Kong sidecar
```
Comment (h3xar0n) [inline app/1.2.x/kubernetes-and-service-mesh.md]:
```suggestion
data from the containers of that pod will flow through a Kong Sidecar.
```

File: app/1.2.x/kubernetes-and-service-mesh.md [hunk 11]
```diff
@@ -0,0 +1,129 @@
...
+
+For example if we utilize the `bookinfo.yaml` example from istio
```
Comment (h3xar0n) [inline app/1.2.x/kubernetes-and-service-mesh.md]:
```suggestion
For example, if we use the `bookinfo.yaml` example from Istio:
```

File: app/1.2.x/kubernetes-and-service-mesh.md [hunk 12]
```diff
@@ -0,0 +1,129 @@
...
+
+```
+kubectl apply -f https://raw.githubusercontent.com/istio/istio/release-1.1/samples/bookinfo/platform/kube/bookinfo.yaml
+```
+
+We see that all pods received a Kong sidecar
```
Comment (h3xar0n) [inline app/1.2.x/kubernetes-and-service-mesh.md]:
```suggestion
We see that all pods received a Kong Sidecar:
```

**Pull request merged**